agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/LitIndexer.java

```diff
@@ -0,0 +1,221 @@
+		
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+	}
+ 
+	
+	@Override
+	protected void startSingleThread(LinkedBlockingDeque<String> queue) {
+	    ObjectMapper mapper = new ObjectMapper();
+	    //String indexName = ConfigHelper.getBlueTeamESIndex();
+	    System.out.println("indexName in startSingleThread:"+indexName);
+	    while (!queue.isEmpty()) {
+	        String page = null;
+	        try {
+	            page = queue.takeFirst();
+	        } catch (InterruptedException e) {
+	            Thread.currentThread().interrupt();
+	            return;
+	        }
+
+	        int maxRetries = 5;
+	        int attempt = 0;
+	        long waitTimeMs = 1000;
+
+	        while (attempt < maxRetries) {
+	            try {
+	                Map<String, Object> object = literatureESApi.search(
+	                	indexName,
+	                    Integer.parseInt(page),
+	                    indexerConfig.getBufferSize()
+	                );
+                    System.out.println("page/from:"+ page + " size: " + indexerConfig.getBufferSize());
+	                Map<String, Object> hitsMap = (Map<String, Object>) object.get("hits");
+	                List<Map<String, Object>> hits = (List<Map<String, Object>>) hitsMap.get("hits");
+
+	                List<LitDocument> list = new ArrayList<>();
+
+	                for (Map<String, Object> map : hits) {
+	                    Map<String, Object> sourceMap = (Map<String, Object>) map.get("_source");
+	                    ObjectNode sourceJson = mapper.convertValue(sourceMap, ObjectNode.class);
+
+	                    // Copy "category" to "literature_category"
+	                    if (sourceJson.has("category")) {
+	                        sourceJson.set("alliance_reference_type", sourceJson.get("category"));
+	                    }
+	                    
+	                    String curie=sourceJson.get("curie").toString();
+		                if (curieSet.contains(curie)) {
+		                    System.out.println("Duplicate curie found: " + curie + " total:"+ curieSet.size());
+		                    System.exit(1);
+		                } else {
+		                    curieSet.add(curie);
+		                }
+
+	                    // Normalize each date field if present
+	                    normalizeAndPutDate(sourceJson, "date_last_modified_in_pubmed");
+	                    normalizeAndPutDate(sourceJson, "date_published");
+	                    normalizeAndPutDate(sourceJson, "date_arrived_in_pubmed");
+	                    normalizeAndPutDate(sourceJson, "date_published_start");
+	                    normalizeAndPutDate(sourceJson, "date_published_end");
+
+	                    LitDocument doc = new LitDocument();
+	                    doc.setObject(sourceJson);
+	                    list.add(doc);
+	                }
+
+	                indexDocuments(list);
+	                break; // success, break out of retry loop
+
+	            } catch (UndeclaredThrowableException e) {
```

Comment:
You should not have to worry about the 429's just run the indexDocuments and devops will take care of the rest.

Reply:
I do some research, seems other suggest to keep it as devops has some limitation to handle this, keeping it will not do any hurt.